yaptide/admin/simulators.py

```diff
@@ -224,6 +240,109 @@ def download_shieldhit_from_s3(
     return True
 
 
+# skipcq: PY-R1000
+def download_topas_from_s3(bucket: str = topas_bucket_name,
+                           key: str = topas_key,
+                           version: str = topas_version,
+                           geant_bucket: str = geant_bucket_name,
+                           path: Path = installation_path
+                           ) -> bool:
+    """Download TOPAS from S3 bucket"""
+    s3_client = boto3.client(
+        "s3",
+        aws_access_key_id=access_key,
+        aws_secret_access_key=secret_key,
+        endpoint_url=endpoint
+    )
+
+    # Download TOPAS tar
+    topas_temp_file = tempfile.NamedTemporaryFile()
+    try:
+        response = s3_client.list_object_versions(
+            Bucket=bucket,
+            Prefix=key,
+        )
+        for curr_version in response["Versions"]:
+            version_id = curr_version["VersionId"]
+            tags = s3_client.get_object_tagging(
+                Bucket=bucket,
+                Key=key,
+                VersionId=version_id,
+            )
+            for tag in tags["TagSet"]:
+                if tag["Key"] == "version" and tag["Value"] == version:
+                    s3_client.download_fileobj(Bucket=bucket,
+                                               Key=key,
+                                               Fileobj=topas_temp_file,
+                                               ExtraArgs={"VersionId": version_id})
```

Comment:
Add here some infromative messages using click echo, to show the users whats happending, especially related to the versions